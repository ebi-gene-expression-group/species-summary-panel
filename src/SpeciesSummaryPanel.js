import React from 'react'
import PropTypes from 'prop-types'

import { CarouselCardsRow } from '@ebi-gene-expression-group/atlas-homepage-cards'

const tabsId = `species-summary-tabs`

class SpeciesSummaryPanel extends React.Component {
  render () {
    const { speciesSummary, carouselCardsRowProps } = this.props

    return (
      [
        <ul key={`tabs`} className={`tabs`} data-tabs id={tabsId}>
          {
            speciesSummary.map(({ kingdom }, idx) =>
              <li key={idx} className={`tabs-title${idx === 0 ? ` is-active` : ``}`} style={{ textTransform: `capitalize` }}>
                <a href={`#${kingdom}`}>{kingdom}</a>
              </li>
            )
          }
        </ul>,
        <div key={`tabs-content`} className={`tabs-content`} data-tabs-content={tabsId}>
          {
            speciesSummary.map(({ kingdom, cards }, idx) =>
              <div key={idx} className={`tabs-panel${idx === 0 ? ` is-active` : ``}`} id={kingdom}>
                <CarouselCardsRow
                  cards={cards}
                  {...carouselCardsRowProps}
                />
              </div>
            )
          }
        </div>
      ]
    )
  }

  // Since the tabs need $().foundation() to function properly, we provide an optional callback to be run on the
  // target DOM node once the component has been initially rendered
  componentDidMount () {
    this.props.onComponentDidMount()
  }
}

SpeciesSummaryPanel.propTypes = {
  speciesSummary: PropTypes.arrayOf(PropTypes.shape({
    kingdom: PropTypes.string.isRequired,
    cards: CarouselCardsRow.propTypes.cards
  })).isRequired,
  onComponentDidMount: PropTypes.func,
  carouselCardsRowProps: PropTypes.object
}

SpeciesSummaryPanel.defaultProps = {
  onComponentDidMount: () => {},
  responsiveCardsRowProps: {}
}

export default SpeciesSummaryPanel
