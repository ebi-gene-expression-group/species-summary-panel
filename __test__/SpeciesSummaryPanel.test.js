/**
 * @jest-environment jsdom
 */

import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'

import SpeciesSummaryPanel from '../src/SpeciesSummaryPanel'

import { speciesSummary } from './TestUtils'

// A side benefit of mocking the cards is that we donâ€™t need to mock window.matchMedia, which is used by the carousel
jest.mock(
  `@ebi-gene-expression-group/atlas-homepage-cards`,
  () => {
    const AtlasHomepageCardsMock = {
      CarouselCardsRow: function (props) {
        return (
          <div></div>
        )
      }
    }
    AtlasHomepageCardsMock.CarouselCardsRow.propTypes = {
      cards: function () {}
    }

    return AtlasHomepageCardsMock
  }
)

const carouselCardsRowProps = {
  className: `row expanded small-up-2 medium-up-3 large-up-6`,
  cardContainerClassName: `column`,
  sliderSettings: { slidesToShow: 2 },
  containerHeight: `320px`,
  sliderHeight: `300px`
}

describe(`SpeciesSummaryPanel`, () => {
  test(`runs callback after mounting`, () => {
    const container = document.createElement(`div`)
    document.body.appendChild(container)

    const mockCallback = jest.fn()
    act(() => {
      ReactDOM.render(
        <SpeciesSummaryPanel
          {...carouselCardsRowProps}
          speciesSummary={[]}
          onComponentDidMount={mockCallback}
        />,
        container)
    })

    expect(mockCallback.mock.calls.length).toBe(1)
  })

  test(`matches snapshot with something to show`, () => {
    const tree = renderer
      .create(<SpeciesSummaryPanel
        carouselCardsRowProps={carouselCardsRowProps}
        {...speciesSummary}
      />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`matches snapshot with nothing to show`, () => {
    const tree = renderer
      .create(<SpeciesSummaryPanel
        carouselCardsRowProps={carouselCardsRowProps}
        speciesSummary={[]}
      />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
