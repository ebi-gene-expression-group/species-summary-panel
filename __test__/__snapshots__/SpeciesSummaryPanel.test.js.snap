// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeciesSummaryPanel matches snapshot with nothing to show 1`] = `
Array [
  <ul
    className="tabs"
    data-tabs={true}
    id="species-summary-tabs"
  />,
  <div
    className="tabs-content"
    data-tabs-content="species-summary-tabs"
  />,
]
`;

exports[`SpeciesSummaryPanel matches snapshot with something to show 1`] = `
Array [
  <ul
    className="tabs"
    data-tabs={true}
    id="species-summary-tabs"
  >
    <li
      className="tabs-title is-active"
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      <a
        href="#animals"
      >
        animals
      </a>
    </li>
    <li
      className="tabs-title"
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      <a
        href="#plants"
      >
        plants
      </a>
    </li>
    <li
      className="tabs-title"
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      <a
        href="#fungi"
      >
        fungi
      </a>
    </li>
  </ul>,
  <div
    className="tabs-content"
    data-tabs-content="species-summary-tabs"
  >
    <div
      className="tabs-panel is-active"
      id="animals"
    >
      <div />
    </div>
    <div
      className="tabs-panel"
      id="plants"
    >
      <div />
    </div>
    <div
      className="tabs-panel"
      id="fungi"
    >
      <div />
    </div>
  </div>,
]
`;
